src/Microsoft.Xaml.Behaviors/NameResolver.cs

```diff
@@ -37,7 +37,7 @@ public NameResolvedEventArgs(object oldObject, object newObject)
     /// Helper class to handle the logic of resolving a TargetName into a Target element
     /// based on the context provided by a host element.
     /// </summary>
-    internal sealed class NameResolver
+    public sealed class NameResolver
```

Comment:
Same here, why public?

Reply:
An example of its use can be found in https://github.com/MartinEgli/XamlBehaviorsWpf/tree/Behaviors.Extensions/src/Behaviors.Extensions . The extension is in a separate project. A class similar to EventTriggerBase should be created that inherits from TriggerBase and the class should behave similarly. Access to internal elements is required to create a class DependencyPropertyChangedEventTriggerBase. 

The class DependencyPropertyChangedEventTriggerBase also needs access to the NameResolvers.